File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        flowers = len(flowerbed)
        if flowers == 1:
            if (n == 0) or ((n == 1) and (flowerbed[0] == 0)):
                return True
            return False
        for i in range(flowers):
            if n < 1:
                break
            if flowerbed[i] == 0:
                if ((i == 0) and (flowerbed[1] == 0)) or ((i == (flowers - 1)) and (flowers >= 2) and (flowerbed[flowers-2] == 0)) or ((flowerbed[i-1] == 0) and (flowerbed[i+1] == 0)):
                    flowerbed[i] = 1
                    n -= 1
        return (n < 1)

File: 0605-can-place-flowers/0605-can-place-flowers.rs
impl Solution {
    pub fn can_place_flowers(flowerbed: Vec<i32>, n: i32) -> bool {
        let mut nb_to_place: i32 = n;
        let flowers: i32 = flowerbed.len() as i32;
        if (flowers == 1) {
            if ((nb_to_place == 0) || ((nb_to_place == 1) && (flowerbed[0] == 0))) {
                return true;
            }
            return false;
        }
        let mut f_bed: Vec<i32> = Vec::from(flowerbed);  // mutable version
        for i in 0..flowers as usize {
            if (nb_to_place < 1) {
                break;
            }
            if (f_bed[i] == 0) {
                if (i == 0) {
                    if (f_bed[1] == 0) {
                        f_bed[i] = 1;
                        nb_to_place -= 1;
                        if (nb_to_place < 1) {
                            break
                        }
                    }
                }
                else if (i == ((flowers as usize) - 1)) {
                    if (f_bed[(flowers as usize)-2] == 0) {
                        f_bed[i] = 1;
                        nb_to_place -= 1;
                        if (nb_to_place < 1) {
                            break
                        }
                    }
                }
                else if ((f_bed[i-1] == 0) && (f_bed[i+1] == 0)) {
                    f_bed[i] = 1;
                    nb_to_place -= 1;
                    if (nb_to_place < 1) {
                        break
                    }
                }
            }
        }
        return (nb_to_place < 1);
    }
}
